sklearn/inspection/_plot/tests/test_plot_partial_dependence.py

```diff
@@ -22,23 +22,28 @@
 
 
 @pytest.fixture(scope="module")
-def boston():
-    return load_boston()
+def diabetes():
+    diabetes = load_diabetes()
+    # Match type of `feature_names` of prev used `load_boston`
+    diabetes.feature_names = np.asarray(diabetes.feature_names)
```

Comment:
Is there somewhere in this file that requires this to be a ndarray?

Reply:
I decided to change the type of `feature_names` instead of changing the tests, but I can change the test instead:
* [test_plot_partial_dependence_str_features](https://github.com/scikit-learn/scikit-learn/blob/e2256c111d7ad7a3cd8e03cd5b12250f087cbdb3/sklearn/inspection/_plot/tests/test_plot_partial_dependence.py#L113) cycles through various `feature_names` types. Can replace list with array here in the parameterization.

Also [`test_plot_partial_dependence_custom_axes`](https://github.com/scikit-learn/scikit-learn/blob/e2256c111d7ad7a3cd8e03cd5b12250f087cbdb3/sklearn/inspection/_plot/tests/test_plot_partial_dependence.py#L177) specifically converts `feature_names` to a list but I didn't understand what it is testing/why this would make a difference. Thus I didn't want to change the original type of `feature_names` in case it makes a difference that I don't understand (if that makes sense).